lib/hotel_booker.rb

```diff
@@ -0,0 +1,139 @@
+require 'date'
+require_relative 'room'
+require_relative 'reservation'
+require_relative 'date_range'
+
+module Hotel
+  class HotelBooker
+    attr_accessor :rooms, :reservations, :unreserved_block, :reserved_block
+    NUM_ROOMS = 20
+    BLOCK_MAX = 5
+
+    def initialize()
+      @unreserved_block = []
+      @reserved_block = []
+      @reservations = []
+      @rooms = []
+
+      NUM_ROOMS.times do |i|
+        @rooms << Hotel::Room.new(i+1)
```

Comment:
Ruby will be looking for names locally first, prioritizing local names... so, since they're all in the same module, there's no need to namespace `Hotel::Room`... `Room` should just work! Same with the rest of the namespacing in the code.